lib/bolt/applicator.rb

```diff
@@ -217,6 +217,7 @@ def apply_ast(raw_ast, targets, options, plan_vars = {})
       r = @executor.log_action(description, targets) do
         futures = targets.map do |target|
           Concurrent::Future.execute(executor: @pool) do
+            Thread.current[:name] ||= Thread.current.name
```

Comment:
What is this doing? I'm assuming the `:name` attribute is the same as the name returned from the name getter, though the docs don't explicitly say so.

Reply:
`:name` is just a thread local variable, separate from actual thread "name". concurrent-ruby sets `thread.name` when it makes the worker threads, but the logging library looks up `thread[:name]` when you use `%T` in the log formatter. So this is copying the name set by concurrent-ruby to the location used by the logging library.

Reply:
Ha! TIL. Thanks for the explanation!